### ISSUE TITLE
targeted commands

### ISSUE THREAD

**Issue by bmorel** [Unvanquished/Unvanquished#2074]
Currently, when you run a command like `/kick` you need to fill in the player's name or number, and there's no easy way for that.
It would be nice to let admins use the crosshair to give the command a target, like what `/navcon start` does, for example.

**Comment by DolceTriade** [comment]
How would you imagine the UX working in this case? Is a UI for admin functions a prerequisite for this? I imagine it would be odd to run an admin command like `/kick -` from the console and then have to close your console and then try to aim at a player... It seems the right thing to do would be create a drop-down menu or something.

A more general issue is a UI and better UX for admin commands than just a CLI

**Comment by bmorel** [comment]
> it would be odd to run an admin command like /kick - from the console and then have to close your console and then try to aim at a player...

Obviously, it would be weird. The idea is really about avoiding using the CLI for that, as it's a real pain for me. Seriously, `listplayers` when it works, then filter out the bots (if you want a kick or something) and finally type the command + ID. And hopefully the person is still playing around and game has not ended in the meantime...

Maybe we can have a circle menu that let the admin choose a "weapon", and implement some kind of "laser" to point at a target?
So, circle menu to select the action, this replaces your weapon with a laser, attack changes the colour of laser, when you hit someone, you get a popup or something, so that you can confirm (or not) the kick/ban/slap/whatever?
This seems reasonably safe and easy to me.

There might also be a need to apply the action to the player you're following, though.

**Comment by DolceTriade** [comment]
I like the idea of an admin menu. (Assuming an admin is a spec, they don't need to move their mouse, so the loss of ingame focus isn't a blocker, and then they can pre-select a player they are interested in, or default the current followed player.

WDYT about renaming this bug to `CLI-less admin UX`

**Comment by bmorel** [comment]
> Assuming an admin is a spec

Problem is, admins are not always spec, sometimes (often, I hope!) they play.

**Comment by DolceTriade** [comment]
One more thing to note is, admins have a /pause function to stop bad people from doing things to allow them time to handle the situation. I think it is extremely difficult to admin and play at the same time (ie, how do they admin what's happening on the other team?)

**Comment by bmorel** [comment]
true. But when game is paused, admins can't move. Also, if you go spectate, you *loose* your credits and equipment, so this is hardly a solution.

**Comment by slipher** [comment]
If we had something like the scoreboard but allowing the mouse to be used, you could right-click on players for admin commands and other things. Vote to kick/mute/denybuild, admin actions for the same things, or send private message.

**Comment by slipher** [comment]
Maybe double-tapping +scores could bring up the mouse-enabled version of the scoreboard. We apparently still transmit information about which buttons have been double-tapped in usercmd_t.

**Comment by illwieckz** [comment]
I also wondered if touble-tapping tab would list the numbers and names in console if it doesn't already do.

**Comment by DolceTriade** [comment]
I personally like the UX that hold tab to show a scoreboard, but hold tab and right click to make the scoreboard clickable.

**Comment by bmorel** [comment]
Agree with the scoreboard.

**Comment by slipher** [comment]
Right click may break use cases of looking at the scoreboard while you are walking around. If you are goon then moving around involves right click

**Comment by DolceTriade** [comment]
Fair point. One other thing that might be more of an implementation detail is that the scoreboard is part of the HUD, but the HUD does not get any inputs (ie, key, mouse, etc) because that would complicate focus issues. That means any interactable UI *must* be in the menu context. that means that any action we do to trigger the menu must end up in the menu context.
